tools/packaging/debian/dragonfly.preinst

```diff
@@ -0,0 +1,55 @@
+#!/bin/sh
+
+set -eu
+# Script to run before the installation starts.
+# We are creating a user "dragonfly", and the directories that
+# would be used by the application
+
+USER="dragonfly"
+
+Setup_dir () {
+	DIR="${1}"
+	MODE="${2}"
+	GROUP="${3}"
+	echo "setting up dir ${DIR} for group ${GROUP}"
+
+	mkdir -p ${DIR} || {
+		echo "failed to create dir ${DIR}"
+		return 1
+	}
+
+	if ! dpkg-statoverride --list ${DIR} >/dev/null 2>&1
+	then
+		echo "changing owner for ${DIR} to user ${USER}"
+		chown ${USER}:${GROUP} ${DIR}
+		chmod ${MODE} ${DIR}
+	fi
+}
+
+echo "running preintall script with $1"
+
+if [ "$1" = "install" ]; then
+	if ! id ${USER} >/dev/null 2>&1 ; then
+		echo "trying to create user ${USER}"
+		adduser \
+			--system \
+			--home /var/lib/${USER} \
+			--quiet \
+			--group \
+			${USER} || {
+			echo "failed to add user ${USER}"
+			exit 1
+		}
+
+		Setup_dir /var/log/${USER} 2755 adm
+		Setup_dir /var/lib/${USER} 755 ${USER}
+		Setup_dir /var/run/${USER} 755 ${USER}
+		Setup_dir /etc/${USER} 2775 ${USER}
+	else
+		echo "user ${USER} already exists"
+		grep ${USER} /etc/passwd
+	fi
+fi
+#DEBHELPER#
```

Comment:
how DEBHELPER is used exactly?
also, how moreutils and dpkg-dev are used as well?

Reply:
the dehelper is the package that actually build from the files (the is what rules file are for), and moreutils is to  automate the changelog from git (so we don't need to write it).